src/pages/index.tsx

```diff
@@ -28,15 +28,48 @@ const HomePage: NextPageWithLayout = () => {
   const [componentProps, setComponentProps] = useState<Record<string, unknown>>({});
   const [showNotificationModalState, setShowNotificationModalState] = useState(false);
   const accountId = useAuthStore((store) => store.accountId);
+  const [tosData, setTosData] = useState<TosData | null>(null);
+  const cacheTosData = useMemo(() => tosData, [tosData?.latestTosVersion]);
+
+  const handleModalCloseOnEsc = useCallback(() => {
+    setShowNotificationModalState(false);
+  }, []);
+
+  const turnNotificationsOn = useCallback(
+    () =>
+      handleTurnOn(accountId, () => {
+        setShowNotificationModalState(false);
+      }),
+    [],
+  );
+
+  const pauseNotifications = useCallback(() => {
+    handleOnCancel();
+    setShowNotificationModalState(false);
+  }, []);
+
+  const checkNotificationModal = useCallback(() => {
+    if (tosData && tosData.agreementsForUser.length > 0) {
+      // show notification modal for new users
+      const tosAccepted =
+        tosData.agreementsForUser[tosData.agreementsForUser.length - 1].value === tosData.latestTosVersion;
+      // check if user has already turned on notifications
+      const { showOnTS } = getNotificationLocalStorage();
+
+      if ((tosAccepted && !showOnTS) || (tosAccepted && showOnTS < Date.now())) {
```

Comment:
`showOnTS < Date.now()` isn't this guaranteed to be true?

Reply:
ah nevermind. I see https://github.com/near/near-discovery/blob/fe8b7e7a1ee50b1cdbfaac62baeba7f8e817e4bb/src/utils/notificationsLocalStorage.ts#L15